src/app/icd/client/CheckInDelegate.h

```diff
@@ -39,6 +41,25 @@ class DLL_EXPORT CheckInDelegate
                                node that sent the check-in message.
      */
     virtual void OnCheckInComplete(const ICDClientInfo & clientInfo) = 0;
+
+    /**
+     * @brief Callback used to let the application know that a key refresh is
+     * needed to avoid counter rollover problems.
+     *
+     * The implementer of this function should generate a new key and store it in a map with ScopedNodeID as the key and
+     * ICDRefreshKeyInfo as the value.
```

Comment:
map is implementation detail, and should not mention in delegate.h, application would figure out how to store it.

Reply:
We don't use map anymore even internally. I've upadted the description.